### ISSUE TITLE
unit testing and profiling

### ISSUE THREAD

**Issue by jhlandman** [quantgroup/laplace_breadth#4]
On anyone's recommendation, should I start with unit testing and profiling right away? Or are we going to do that at a later development date?

Also is it better to do profiling on the server-side where the calculation is going to be performed?

**Comment by nmarshall** [comment]
I don't think it is ever too early to start on the unit test aspect. Python has a built in `unittest` module however I think [pytest] [https://docs.pytest.org/en/latest/](https://docs.pytest.org/en/latest/) has some pretty good reviews. 

Maybe ask @zachzIAM what he has used while working with Yingke. Pytest has a number of plugins including one for profiling [https://pypi.org/project/pytest-profiling/](https://pypi.org/project/pytest-profiling/) which may help
